source/fuzz/transformation_swap_functions.h

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2021 Emiljano Gjiriti
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SOURCE_FUZZ_TRANSFORMATION_PERMUTE_FUNCTIONS_H_
+#define SOURCE_FUZZ_TRANSFORMATION_PERMUTE_FUNCTION_H_
+
+#include "source/fuzz/protobufs/spirvfuzz_protobufs.h"
+#include "source/fuzz/transformation.h"
+#include "source/fuzz/transformation_context.h"
+#include "source/opt/ir_context.h"
+
+namespace spvtools {
+namespace fuzz {
+
+class TransformationSwapFunctions : public Transformation {
+ public:
+  explicit TransformationSwapFunctions(
+      protobufs::TransformationSwapFunctions message);
+
+  TransformationSwapFunctions(uint32_t func1_id, uint32_t func2_id);
+
+  // - |permutation| is a set of [0..(n - 1)], where n is the number
+  //   of functions in the module
```

Comment:
Docs to `IsApplicable` usually describe the preconditions you checked in that method. In our case:
- Both result ids must be valid result ids of some functions in the module.
- They can't be equal.